pyblp/economies/economy.py

```diff
@@ -73,11 +73,8 @@ def __init__(
         self.H = self.unique_nesting_ids.size
 
         # identify market indices
-        self._product_market_indices: Dict[Hashable, Array] = {}
-        self._agent_market_indices: Dict[Hashable, Array] = {}
-        for t in self.unique_market_ids:
-            self._product_market_indices[t] = np.flatnonzero(self.products.market_ids == t)
-            self._agent_market_indices[t] = np.flatnonzero(self.agents.market_ids == t)
+        self._product_market_indices: Dict[Hashable, Array] = get_indices(self.products.market_ids)
+        self._agent_market_indices: Dict[Hashable, Array] = get_indices(self.agents.market_ids)
```

Comment:
You can remove the type annotations here. They're now inferred by the return type of `get_indices`.